dotCMS/src/main/java/com/dotmarketing/portlets/personas/business/MultiTreePersonaListener.java

```diff
@@ -0,0 +1,70 @@
+package com.dotmarketing.portlets.personas.business;
+
+import com.dotcms.content.elasticsearch.business.event.ContentletDeletedEvent;
+import com.dotcms.content.elasticsearch.business.event.ContentletPublishEvent;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Versionable;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ContentletListener;
+import com.dotmarketing.portlets.personas.model.Persona;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.util.StringPool;
+
+import java.util.Optional;
+
+/**
+ * This listener is to track the changes on personas in order to keep up to date changes on the
+ * the personalization multitree
+ * By default it is just cleaning up when the publish/unpublish or delete
+ * @author jsanca
+ */
+public class MultiTreePersonaListener implements ContentletListener<Persona> {
+
+
+    @Override
+    public void onModified(final ContentletPublishEvent<Persona> contentletPublishEvent) {
+
+        try {
+
+            final Optional<Versionable> versionable = APILocator.getVersionableAPI()
+                    .findPreviousVersion(contentletPublishEvent.getContentlet().getIdentifier());
+
+            if (versionable.isPresent()) {
```

Comment:
Should we check if the persona is live or not?  We could have an working version with `persona:ABC` and the previous version (live) has `persona:DEF`.  Changing personalization to the value of a working version would basically remove any personalization for the live version.